src/game.cpp

```diff
@@ -14561,11 +14561,11 @@ void game::start_calendar()
         if( scen->has_flag("SPR_START") ) {
             ; // Do nothing;
         } else if( scen->has_flag("SUM_START") ) {
-            calendar::start += DAYS((int)ACTIVE_WORLD_OPTIONS["SEASON_LENGTH"]);
+            calendar::start += DAYS( calendar::season_length() );
```

Comment:
I removed `set_season_length`, `season_length` now gets its values directly from the world options.